src/app/workspace/workspace-chooser/workspace-chooser.component.ts

```diff
@@ -72,37 +79,94 @@ export class WorkspaceChooserComponent implements OnInit {
-            .subscribe((projects: Project[]) => {
-              activeOrganization.projects = projects;
-
-              if (this.workspace.projectCode) {
-                const ixProj: number = activeOrganization.projects.findIndex(proj =>
-                  proj.code === this.workspace.projectCode
-                );
-                if (ixProj >= 0) {
-                  this.activeProjIx = ixProj;
+    Observable.combineLatest(
+      this.store.select(selectWorkspace),
+      this.store.select(selectOrganizationsState),
+      this.store.select(selectProjectsState)
+    ).pipe(
+      first()
+    ).subscribe(([workspace, organizationsState, projectsState]) => {
+      this.workspace = workspace;
+      this.organizationsState = organizationsState;
+      this.projectsState = projectsState;
+
+      this.getOrganizations();
+    });
+  }
+
+  private getOrganizations(): void {
+    this.organizationService.getOrganizations().subscribe(
+      organizations => {
+        this.organizations = organizations;
+
+        if (this.workspace.organizationCode) {
+          const ix: number = this.organizations.findIndex(org =>
+            org.code === this.workspace.organizationCode
+          );
+          if (ix >= 0) {
+            this.activeOrgIx = ix;
+
+            const activeOrganization = this.organizations[this.activeOrgIx];
+            activeOrganization.projects = [];
+            this.projectService.getProjects(activeOrganization.code)
+              .subscribe((projects: Project[]) => {
+                activeOrganization.projects = projects;
+
+                if (this.workspace.projectCode) {
+                  const ixProj: number = activeOrganization.projects.findIndex(proj =>
+                    proj.code === this.workspace.projectCode
+                  );
+                  if (ixProj >= 0) {
+                    this.activeProjIx = ixProj;
+                  }
                 }
-              }
-            });
+              });
+          }
+        }
+
+        this.selectPreviousWorkspace();
+      });
+  }
+
+  private selectPreviousWorkspace(): void {
+    const previouslySelectedOrganizationIndex = this.getPreviouslySelectedOrganizationIndex();
+    if (previouslySelectedOrganizationIndex !== -1) {
+      this.activeOrgIx = previouslySelectedOrganizationIndex;
+      this.selectPreviouslySelectedProject(previouslySelectedOrganizationIndex);
+    }
+  }
+
+  private getPreviouslySelectedOrganizationIndex(): number {
+    const selectedOrganizationCode = this.organizationsState.selectedOrganizationCode;
+    return this.organizations.findIndex(organization => organization.code === selectedOrganizationCode);
+  }
+
+  private selectPreviouslySelectedProject(index: number) {
+    const selectedOrganization = this.organizations[index];
+    this.getOrganizationProjects(selectedOrganization).subscribe(
+      projects => {
+        selectedOrganization.projects = projects;
```

Comment:
This is against NgRx principles of having all objects in the store immutable. If you want to change anything, you need to dispatch the right action. The object in the store can only be changed in the reducer function!

There has already been created an action to select a given project.